ui/page/root/home_page.go

```diff
@@ -219,6 +220,79 @@ func (hp *HomePage) initDEX() {
 		dexPassEditor := hp.Theme.EditorPassword(new(widget.Editor), values.String(values.StrDexPassword))
 		dexPassEditor.Editor.SingleLine, dexPassEditor.IsRequired = true, true
 
+		showWalletToSyncModal := func() {
+			// DEX client has active orders or expired bonds, retrieve the
+			// wallets involved and ensure they are synced or syncing.
+			walletsToSyncMap := make(map[uint32]*struct{})
+			for _, orders := range activeOrders {
+				for _, ord := range orders {
+					walletsToSyncMap[ord.BaseID] = &struct{}{}
+					walletsToSyncMap[ord.QuoteID] = &struct{}{}
+				}
+			}
+
+			for _, bond := range expiredBonds {
+				walletsToSyncMap[bond.AssetID] = &struct{}{}
+			}
+
+			var walletsToSyncStr string
+			var walletsToSync []sharedW.Asset
+			for assetID := range walletsToSyncMap {
+				walletID, err := dexClient.WalletIDForAsset(assetID)
+				if err != nil {
+					log.Errorf("dexClient.WalletIDForAsset(%d) error: %w", assetID, err)
+					continue
+				}
+
+				if walletID == nil {
+					continue // impossible but better safe than sorry
+				}
+
+				wallet := hp.AssetsManager.WalletWithID(*walletID)
+				if wallet == nil { // impossible but better safe than sorry
+					log.Error("dex wallet with ID %d is missing", walletID)
+					continue
+				}
+
+				if wallet.IsSynced() || wallet.IsSyncing() {
+					continue // ok
+				}
+
+				walletsToSync = append(walletsToSync, wallet)
+				walletsToSyncStr += "," + fmt.Sprintf("%s (%s)", wallet.GetWalletName(), wallet.GetAssetType())
+			}
+
+			if len(walletsToSync) == 0 {
+				return
+			}
+
+			walletSyncRequestModal := modal.NewCustomModal(hp.Load).
+				Title(values.String(values.StrWalletsNeedToSync)).
+				Body(values.StringF(values.StrWalletsNeedToSyncMsg, strings.Trim(walletsToSyncStr, ","))).
+				SetNegativeButtonText(values.String(values.StrIWillSyncLater)).
+				SetPositiveButtonText(values.String(values.StrOkaySync)).
+				SetPositiveButtonCallback(func(isChecked bool, im *modal.InfoModal) bool {
+					if !hp.isConnected.Load() {
+						// Notify user and return.
+						hp.Toast.NotifyError(values.String(values.StrNotConnected))
+						return false
+					}
+
+					for _, w := range walletsToSync {
+						err := w.SpvSync()
+						if err != nil {
+							log.Error(err)
+						} else {
+							w.SaveUserConfigValue(sharedW.AutoSyncConfigKey, true)
+						}
+					}
+
+					hp.ParentWindow().Reload()
+					return true
```

Comment:
Is the window reload necessary here? When you return true, it'll dismiss the modal which triggers a window reload.

Reply:
Uhmm, I can remove I think. The sync status value is now fetched db right when the `layout` is drawn.